### PR TITLE
Optional bandwidth tracker

### PR DESCRIPTION
**Pull request by MashinaMashina**
Not everyone needs a tracker. If you needs, enable using option:
```
tls_client.WithBandwidthTracker()
```

### CODE REVIEW

File: client_options.go [hunk 1]
```diff
@@ -241,3 +242,9 @@ func WithDisableIPV6() HttpClientOption {
 		config.disableIPV6 = true
 	}
 }
+
+func WithBandwidthTracker(enabled bool) HttpClientOption {
```
Comment (bogdanfinn) [inline client_options.go:246]: This option should not have a parameter in my opinion. If someone wants to enable the Tracker he can add the option (which results in `config.enabledBandwidthTracker = true`) and if you do not want the tracker ommit the option completely.

Similar to the `WithDisableIPV6` option.

**Review by bogdanfinn — CHANGES_REQUESTED** [review]

File: roundtripper.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func (rt *roundTripper) getTransport(req *http.Request, addr string) error {
 		return fmt.Errorf("invalid URL scheme: [%v]", req.URL.Scheme)
 	}
 
-	_, err := rt.dialTLS(context.Background(), "tcp", addr)
```
Comment (justhyped) [inline roundtripper.go:103]: what's this change for?

**Comment by MashinaMashina** [comment]
ctx in TrackConnection method allows add your logic in custom implementation. Example:
```
req, err := http.NewRequestWithContext(context.WithValue(context.Background(), "do_not_track", "1"), http.MethodGet, "https://www.topps.com/", nil)
...
```

```
func (bt *Tracker) TrackConnection(ctx context.Context, conn net.Conn) net.Conn {
	if ctx.Value("do_not_track") != nil {
		return conn
	}

	return newTrackedConn(conn, bt)
}
```

File: roundtripper.go (hunk 2 above)
Comment (MashinaMashina) [reply roundtripper.go:103]: It's just that now we are losing the context between NewRequestWithContext and the request. I wrote next message with example

**Comment by bogdanfinn** [comment]
@MashinaMashina thank you for your contribution

**Pull request merged**